go/ql/lib/semmle/go/security/IncorrectIntegerConversionLib.qll

```diff
@@ -1,16 +1,47 @@
 import go
 
-/**
- * Gets the maximum value of an integer (signed if `isSigned`
- * is true, unsigned otherwise) with `bitSize` bits.
- */
-float getMaxIntValue(int bitSize, boolean isSigned) {
-  bitSize in [8, 16, 32] and
-  (
-    isSigned = true and result = 2.pow(bitSize - 1) - 1
-    or
-    isSigned = false and result = 2.pow(bitSize) - 1
-  )
+/** The constant `math.MaxInt` or the constant `math.MaxUint`. */
+abstract private class MaxIntOrMaxUint extends DeclaredConstant {
+  /**
+   * Gets the integer `x` such that `2.pow(x) - 1` is the value of this
+   * constant when the architecture has bit size `architectureBitSize`.
+   */
+  abstract int getX(int architectureBitSize);
```

Comment:
`getOrder`, as in (binary) order of magnitude?

Reply:
Sure. Pretty much anything is better than `getX` 😆 .